Utilities/CreateServicePrincipal.ps1

```diff
@@ -223,16 +223,34 @@ function New-ServicePrincipal() {
     Write-Prompt "2) Owner"
     Write-Prompt "Please choose by entering 1 or 2:"
     $Privilege = Read-Host
-
     Write-Host "Assign roles to Service Principal..."
-    if ($Privilege -eq 2) {
-        New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId $ClientId
-    }
-    else {
-        New-AzRoleAssignment -RoleDefinitionName "Contributor" -ApplicationId $ClientId
+    try {
+        if ($Privilege -eq 2) {
+            New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId $ClientId
+        }
+        else {
+            New-AzRoleAssignment -RoleDefinitionName "Contributor" -ApplicationId $ClientId
+        }
+        Write-Host "Successfully created Service Principal and assigned role...`n"
+    } catch {
+        $ErrorMessage = $_.Exception.Message
+        Write-Host -ForegroundColor Red "$ErrorMessage"
+        $currentSignInAccount = Get-AzContext | Select-Object Account
+        $role = Get-AzRoleAssignment -SignInName $currentSignInAccount.Account.id | Select-Object RoleDefinitionName
+        # https://docs.microsoft.com/en-us/powershell/azure/create-azure-service-principal-azureps?view=azps-2.5.0#manage-service-principal-roles
+        if ($ErrorMessage -match "does not have authorization to perform action 'Microsoft.Authorization/roleAssignments/write' over scope") {
+            Write-Host "You are the $($role.RoleDefinitionName) of subscription $($subscription.Name) - $($subscription.Id), don't have enough permission to assign a role."
+            $confirmation = Read-Host "Do you want to delete the service principal $identifier just created? [y/n]"
+            if($confirmation -eq "y") {
+                Remove-AzADApplication -ApplicationId $ClientId -Force
+                Write-Host "Delete $identifier successfully..."
+                return
+            } else {
+                Write-Host "Successfully created Service Principal but assigned role failed...`n"
```

Comment:
What is the expected behavior of the user? I don't think we leave the intermediate artifact here.

Reply:
I give the invoke person to make decision, remove it or keep it for further use, he can manually delete it.